symengine/tests/eval/test_eval_double.cpp

```diff
@@ -137,6 +145,23 @@ TEST_CASE("eval_double: eval_double", "[eval_double]")
         REQUIRE(::fabs(val - vec[i].second) < 1e-12);
     }
 
+    // Booleans
+    REQUIRE(eval_double(*boolean(true)) == true);
+    REQUIRE(eval_double(*boolean(false)) == false);
+
+    // Piecewise
+    {
+        RCP<const Basic> x = symbol("x");
+        PiecewiseVec pwv;
+        pwv.push_back({rcp_static_cast<const Basic>(real_double(2.0)),
+                       Gt(x, integer(1))});
+        pwv.push_back(
+            {rcp_static_cast<const Basic>(real_double(3.0)), boolTrue});
+        RCP<const Basic> pw1 = piecewise(std::move(pwv));
+        REQUIRE(eval_double(*subs(pw1, {{x, integer(10)}})) == 2.0);
```

Comment:
Does the `subs` here return a `Piecewise` function? If so, that might be a bug.

Reply:
Yeah, it returned this:
```
Piecewise((2.0, True), (3.0, True))
```

Reply:
Is it a bug because the first condition is always met? I can envision a scenario (e.g. partial substitution) where one would want to return a Piecewise.

Reply:
Yes, a subs could return a Piecewise, but in this case it should return 2.0

Reply:
Ok, thanks for the confirmation!